Move unit tests to GitHub workflows, test also under Python 3.9

nsoranzo opened the pull request:
Also some random fixes to other GitHub workflows.

nsoranzo committed:
Move unit tests to GitHub workflows

Also some random fixes to other GitHub workflows.

mvdbeek commented on .github/workflows/unit.yaml:
```diff
@@ -0,0 +1,28 @@
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6']
```
```suggestion
        python-version: ['3.6', '3.9']
```

nsoranzo replied:
Maybe 3.6 and 3.9, like for lint workflow?

mvdbeek commented on .github/workflows/unit.yaml:
```diff
@@ -0,0 +1,28 @@
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
```suggestion
      fail-fast: false
      matrix:
```

mvdbeek reviewed (approved):
Should we test more python versions ?

nsoranzo committed:
Add Python 3.9 to unit tests CI matrix

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

nsoranzo committed:
Fix deprecation warning

```
lib/galaxy/job_metrics/instrumenters/cgroup.py:28
  /home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/job_metrics/instrumenters/cgroup.py:28: DeprecationWarning: invalid escape sequence \,
    CPU_USAGE_TEMPLATE = """
```

nsoranzo committed:
Remove Python 2 compatibility leftover

nsoranzo committed:
Add Python 3.9 to first_startup CI matrix

nsoranzo committed:
Update isa-rwal and networkx dependencies

Fix the following traceback in unit tests on Python 3.9:

```
_________________ ERROR collecting lib/galaxy/datatypes/isa.py _________________
lib/galaxy/datatypes/isa.py:17: in <module>
    from isatools import (
.tox/unit/lib/python3.9/site-packages/isatools/isajson.py:13: in <module>
    from isatools.model import *
.tox/unit/lib/python3.9/site-packages/isatools/model.py:17: in <module>
    import networkx as nx
.tox/unit/lib/python3.9/site-packages/networkx/__init__.py:84: in <module>
    import networkx.generators
.tox/unit/lib/python3.9/site-packages/networkx/generators/__init__.py:5: in <module>
    from networkx.generators.classic import *
.tox/unit/lib/python3.9/site-packages/networkx/generators/classic.py:21: in <module>
    from networkx.algorithms.bipartite.generators import complete_bipartite_graph
.tox/unit/lib/python3.9/site-packages/networkx/algorithms/__init__.py:12: in <module>
    from networkx.algorithms.dag import *
.tox/unit/lib/python3.9/site-packages/networkx/algorithms/dag.py:2: in <module>
    from fractions import gcd
E   ImportError: cannot import name 'gcd' from 'fractions' (/opt/hostedtoolcache/Python/3.9.1/x64/lib/python3.9/fractions.py)
```
